### PR TITLE
Repo Configuration Fixes

### PR DESCRIPTION
**Pull request by cschuchardt88**
# Change Log
- [x] Fixed reporting with `coverlet.msbuild` 
- [x] Updated `dotnet` version to `8.0.202` in `global.json` file
- [x] Breaking change (fix or feature that would cause existing functionality to not work as expected)

# Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

**Comment by Jim8y** [comment]
version must be specified?

File: .github/workflows/main.yml [hunk 1]
```diff
@@ -35,18 +35,20 @@ jobs:
...
     - name: Coveralls
       if: matrix.os == 'ubuntu-latest'
       uses: coverallsapp/github-action@v2.2.3
       with:
         github-token: ${{ secrets.GITHUB_TOKEN }}
-        format: lcov
-        file: ${GITHUB_WORKSPACE}/coverage/lcov.net8.0.info
+        compare-ref: ${{ github.head_ref }}
+        compare-sha: ${{ github.sha }}
+        base-path: ./TestResults/coverage
```
Comment (Jim8y) [inline .github/workflows/main.yml:51]:
```suggestion
        base-path: ./TestResults/coverage/
```

**Comment by cschuchardt88** [comment]
Working Great Now
👍 🛩️ 💯 🥇 😃 
---
@shargon @Jim8y @vncoelho 

```bash
[coverlet] 
Calculating coverage result...
  [coverlet] MergeWith: '/home/runner/work/neo/neo/TestResults/coverage/coverage.net8.0.json'.
  Generating report '/home/runner/work/neo/neo/TestResults/coverage/coverage.net8.0.info'

+----------------------------+--------+--------+--------+
| Module                     | Line   | Branch | Method |
+----------------------------+--------+--------+--------+
| Neo.Json                   | 84.71% | 65.33% | 81.5%  |
+----------------------------+--------+--------+--------+
| Neo.VM                     | 87.65% | 80.6%  | 88.3%  |
+----------------------------+--------+--------+--------+
| Neo.Cryptography.BLS12_381 | 87.99% | 63%    | 73.96% |
+----------------------------+--------+--------+--------+
| Neo                        | 78.9%  | 65.38% | 81.61% |
+----------------------------+--------+--------+--------+
| Neo.Extensions             | 78.57% | 0%     | 80%    |
+----------------------------+--------+--------+--------+
| Neo.IO                     | 75.36% | 83.82% | 75%    |
+----------------------------+--------+--------+--------+
| Neo.ConsoleService         | 15.66% | 13.04% | 19.4%  |
+----------------------------+--------+--------+--------+
+---------+--------+--------+--------+
|         | Line   | Branch | Method |
+---------+--------+--------+--------+
| Total   | 79.22% | 65.6%  | 80.6%  |
+---------+--------+--------+--------+
| Average | 72.69% | 53.02% | 71.39% |
+---------+--------+--------+--------+
```

**Review by Jim8y — APPROVED** [review]

**Comment by cschuchardt88** [comment]
@Jim8y Sorry I forgot one thing...

File: .github/workflows/main.yml [hunk 2]
```diff
@@ -35,18 +35,19 @@ jobs:
...
+        dotnet test ./tests/Neo.Cryptography.BLS12_381.Tests /p:CollectCoverage=true /p:CoverletOutput='${{ github.workspace }}/TestResults/coverage/'
+        dotnet test ./tests/Neo.ConsoleService.Tests /p:CollectCoverage=true /p:CoverletOutput='${{ github.workspace }}/TestResults/coverage/' /p:MergeWith='${{ github.workspace }}/TestResults/coverage/coverage.net8.0.json'
+        dotnet test ./tests/Neo.UnitTests /p:CollectCoverage=true /p:CoverletOutput='${{ github.workspace }}/TestResults/coverage/' /p:MergeWith='${{ github.workspace }}/TestResults/coverage/coverage.net8.0.json'
+        dotnet test ./tests/Neo.VM.Tests /p:CollectCoverage=true /p:CoverletOutput='${{ github.workspace }}/TestResults/coverage/' /p:MergeWith='${{ github.workspace }}/TestResults/coverage/coverage.net8.0.json'
+        dotnet test ./tests/Neo.Json.UnitTests /p:CollectCoverage=true /p:CoverletOutput='${{ github.workspace }}/TestResults/coverage/'  /p:MergeWith='${{ github.workspace }}/TestResults/coverage/coverage.net8.0.json' /p:CoverletOutputFormat='lcov' -m:1
+        ls -l ${{ github.workspace }}/TestResults/coverage
```
Comment (shargon) [inline .github/workflows/main.yml:43]: We can remove the ls

File: .github/workflows/main.yml [hunk 3]
```diff
@@ -35,18 +35,19 @@ jobs:
...
+        dotnet test ./tests/Neo.Cryptography.BLS12_381.Tests /p:CollectCoverage=true /p:CoverletOutput='${{ github.workspace }}/TestResults/coverage/'
+        dotnet test ./tests/Neo.ConsoleService.Tests /p:CollectCoverage=true /p:CoverletOutput='${{ github.workspace }}/TestResults/coverage/' /p:MergeWith='${{ github.workspace }}/TestResults/coverage/coverage.net8.0.json'
+        dotnet test ./tests/Neo.UnitTests /p:CollectCoverage=true /p:CoverletOutput='${{ github.workspace }}/TestResults/coverage/' /p:MergeWith='${{ github.workspace }}/TestResults/coverage/coverage.net8.0.json'
+        dotnet test ./tests/Neo.VM.Tests /p:CollectCoverage=true /p:CoverletOutput='${{ github.workspace }}/TestResults/coverage/' /p:MergeWith='${{ github.workspace }}/TestResults/coverage/coverage.net8.0.json'
+        dotnet test ./tests/Neo.Json.UnitTests /p:CollectCoverage=true /p:CoverletOutput='${{ github.workspace }}/TestResults/coverage/'  /p:MergeWith='${{ github.workspace }}/TestResults/coverage/coverage.net8.0.json' /p:CoverletOutputFormat='lcov' -m:1
```
Comment (shargon) [inline .github/workflows/main.yml:42]: What means -m 1?

Comment (cschuchardt88) [reply .github/workflows/main.yml:42]: @shargon 
You can merge also running dotnet test and merge with single command from a solution file, but you need to ensure that tests will run sequentially(-m:1). This slow down testing but avoid invalid coverage result.

Comment (vncoelho) [reply .github/workflows/main.yml:42]: Just in the last?

**Review by vncoelho — COMMENTED** [review]
It fixes the coverage increase in https://github.com/neo-project/neo/commit/65822cf81cbea943e3cdaf362ee022d1624fd22d

That is why it decreases now

**Review by vncoelho — COMMENTED** [review]
I think you can remove the `-m:1`

The main fix here was the problem introduced in the last but one PR that forced all formats to be `lcov`

**Review by vncoelho — APPROVED** [review]

**Comment by shargon** [comment]
Lcov havent got the namespace information I think

**Review by AnnaShaleva — APPROVED** [review]
Seems to be legit.

**Review by shargon — APPROVED** [review]

**Pull request merged**